### PR TITLE
Remove `chef_version` user variable & remove Chef installation option.

### PR DESCRIPTION
**Pull request by fnichol**
In an effort to simplify the potential for behavior proliferation that
user variables allows, the `chef_version` user variable is being
dropped. The default behavior was to *not* install Chef for all Bento
boxes as downstream tooling such as Test Kitchen or vagrant-omnibus can
handle Chef installations at boot time.

### CODE REVIEW

**Comment by fnichol** [comment]
@yzl, @scotthain: Can either of you foresee the need the preserve the "install Chef" behavior in Bento?

**Comment by scotthain** [comment]
:+1: I like the boxes agnostic :)

File: centos-5.11-i386.json [hunk 1]
```diff
@@ -102,21 +102,18 @@
   ],
   "post-processors": [
     {
-      "output": "builds/{{.Provider}}/opscode_centos-5.11-i386_chef-{{user `chef_version`}}.box",
+      "output": "builds/{{.Provider}}/opscode_centos-5.11-i386.box",
```
Comment (lamont-granquist) [inline centos-5.11-i386.json]: These kinds of strings are hardcoded here:

https://github.com/test-kitchen/kitchen-vagrant/blob/master/lib/kitchen/driver/vagrant.rb#L112

Needs to still be "_chef-provisionerless.box"

**Comment by lamont-granquist** [comment]
Yeah :+1:, keep these boxes simple and provisonerless.  We don't need to spend time supporting chef and puppet and ansible and salt and that entire bikeshed.

**Comment by yzl** [comment]
:+1:

File: centos-5.11-i386.json (hunk 1 above)
Comment (fnichol) [inline centos-5.11-i386.json]: Ah, that's a good catch. I'm hoping to update kitchen-vagrant just as soon as we can get boxes publishing up to Atlas. That could mean in the very short term that the box would have to be renamed when uploading to the S3 bucket. About 2 PRs down the road, I'm hoping to change the box file names even more so this is going to be important to keep in mind.

**Pull request merged**